yogstation/code/modules/mob/say.dm

```diff
@@ -1,3 +1,29 @@
 /mob/proc/get_say()
+	create_typing_indicator()
 	var/msg = input(src, null, "say \"text\"") as text|null
-	say_verb(msg)
\ No newline at end of file
+	remove_typing_indicator()
+	say_verb(msg)
+
+/mob
+	var/image/typing_overlay
+	var/list/speech_bubble_recipients
+
+/mob/proc/create_typing_indicator()
+	if(typing_overlay) return
+	var/list/listening = get_hearers_in_view(6, src) //we do a little bit of hardcoding
```

Comment:
find a way to not hardcode this then? use widescreen size define or something idk